tests/examples/source/devfiles/springboot/devfile.yaml

```diff
@@ -5,47 +5,37 @@ metadata:
 starterProjects:
   - name: springbootproject
     git:
-      location: "https://github.com/maysunfaisal/springboot.git"
+      location: "https://github.com/odo-devfiles/springboot-ex.git"
 components:
   - container:
       name: tools
-      image: maysunfaisal/springbootbuild
+      image: quay.io/eclipse/che-java11-maven:nightly
       memoryLimit: 768Mi
       command: ['tail']
       args: [ '-f', '/dev/null']
-      volumeMounts:
-        - name: springbootpvc
-          path: /data
       mountSources: true
   - container:
       name: runtime
-      image: maysunfaisal/springbootruntime
+      image: quay.io/eclipse/che-java11-maven:nightly
       memoryLimit: 768Mi
-      command: ['tail']
-      args: [ '-f', '/dev/null']
       endpoints:
         - name: "8080/tcp"
           targetPort: 8080
-      volumeMounts:
-        - name: springbootpvc
-          path: /data
-      mountSources: false
-  - volume:
-      name: springbootpvc
+      mountSources: true
```

Comment:
lets keep this is as false, so that one container has the source while the other doesnt

Reply:
there are lots of issues if we do ` mountSources: false`, we need to copy the jar files from build to run container, tried doing that (check this devfile https://gist.github.com/adisky/d690abc464f65f864baacd47f29635f6) but faced lots of issue related to file permission.

Reply:
I suspect its because you're on an OpenShift cluster? OpenShift starts container as a non-root user and your container's default user does not have permission to write to pvc dirs.

Reply:
The devfile is good but the reason why the previous Spring devfile didnt utilize the project mount was because it is an emptyDir vol, and if for any reason the pod restarts; the contents are lost and that would include the artifacts/jar which runs the Spring server https://kubernetes.io/docs/concepts/storage/volumes/#emptydir This may mean, we would have to do a mvn build again to get the application up and running.

If we're however storing the artifacts/jar in a pvc; pvc contents are always present as long as the deployment is present. If for any reason, the pod restarts; the application would be back up and running.

However, since this a test devfile and we're really testing the two component container logic; it should be ok! Manually tried it out as well. 

/lgtm

Reply:
completely agree that, but it is in general problem with all devfiles, there is separate issue to figure out to save the code/binaries in pvc instead of empty dir.